src/pipdeptree/_models/package.py

```diff
@@ -23,13 +26,9 @@ class Package(ABC):
 
     UNKNOWN_LICENSE_STR = "(Unknown license)"
 
-    def __init__(self, obj: DistInfoDistribution) -> None:
-        self._obj: DistInfoDistribution = obj
-        self.key = pep503_normalize(obj.key)
-
-    @property
-    def project_name(self) -> str:
-        return self._obj.project_name  # type: ignore[no-any-return]
+    def __init__(self, obj: Distribution) -> None:
+        self._obj: Distribution = obj
+        self.key = pep503_normalize(self._obj.metadata["Name"])
```

Comment:
Looking at the patch, I think we could do the following:
```suggestion
    def __init__(self, project_name: str) -> None:
         self.project_name = project_name
        self.key = pep503_normalize(project_name)
```

Let's just get rid of passing `obj` entirely in `Package.__init__()`. You also need to:

- Update `Package.project_name()` to return `self.project_name`
- Have its subclasses pass the project name instead of its object they are wrapping over
- Assign the wrapper objects `Distribution` and `Requirement` to fields in their respective classes (i.e. `DistPackage` and `ReqPackage`) and use them there

This avoids more code duplication; also, it's better than passing objects that aren't related in any way with each other to `Package.__init__()` and treating them like they are elsewhere.

Reply:
Ok, I've done this part. You are right, this can remove duplication of `project_name` in code.